File: packages/types/src/history.ts
export type ExamHistoryItem = {
  id: string;
  resultId: string;
  examId: string;
  examTitle: string;
  takenAt: string;
  questionCount: number;
  totalScore: number;
  maxScore: number;
  durationSeconds: number;
};

export type MyPageProfile = {
  id: string;
  username: string;
  email: string;
  name: string;
  totalExamCount: number;
};

File: apps/api/src/moldo_api/schemas/history.py
from pydantic import BaseModel, ConfigDict


class ApiDto(BaseModel):
    model_config = ConfigDict(extra="forbid")


class ExamHistoryItem(ApiDto):
    id: str
    resultId: str
    examId: str
    examTitle: str
    takenAt: str
    questionCount: int
    totalScore: int
    maxScore: int
    durationSeconds: int


class MyPageProfile(ApiDto):
    id: str
    username: str
    email: str
    name: str
    totalExamCount: int
